src/ir_passes/bundle_elim.rs

```diff
@@ -25,20 +26,28 @@ impl BundleElim {
         let comp = ctx.get(cidx);
 
         let Access { port, start, end } = access;
-
         let start = start.as_concrete(comp).unwrap() as usize;
         let end = end.as_concrete(comp).unwrap() as usize;
-        self.context[cidx][port][start..end].to_vec()
+
+        (start..end)
+            .map(|idx| {
+                let mut group = (*port, idx);
+                let (port, idx) = loop {
+                    match self.local_map.get(&group) {
+                        Some(&g) => group = g,
+                        None => break group,
+                    }
+                };
+                self.context[cidx][&port][idx]
+            })
+            .collect()
```

Comment:
Hm, this is hard to read: In general, mixing iterators and imperative operators like `break` becomes confusing quickly.

Reply:
Is there a better method to this? I have to mix them because I want to collect a vec of ports at the end, and using `for` instead would require creating a mutable vec which I'm not sure is any cleaner.

Reply:
I usually just bite the bullet and write the imperative version of the code and explicitly allocate a vector using `Vec::with_capacity`